### PR TITLE
Fixes for lightning 2.0 upgrade

### PR DESCRIPTION
**Pull request by athitten**
# What does this PR do ?

Fix the following issues with the previous lightning 2.0 upgrade PR:

1) Decrease `val_check_interval` to 1 while resuming from a ckpt in some CI tests where max_steps is same as that of training and lightning 2.0 errors: [fit_loop.py#L259](https://github.com/Lightning-AI/lightning/blob/2.0.6/src/lightning/pytorch/loops/fit_loop.py#L259). Lightning 2.0 calls `setup_data` right at the beginning of fit_loop.run unlike 1.9 that did it only when `self.skip` for fit was False, hence the error din't show up with 1.9
2) Remove the usage of `trainer._checkpoint_connector` as it's redundant in lightning 2.0 and use `trainer.ckpt_path` directly. Also having `trainer._checkpoint_connector = _CheckpointConnector(trainer)` overrode ckpt_path with None leading to starting training from scratch during resuming from a ckpt.
3) Remove `optimizer_idx` arg in `optimizer_step` of `MegatronHalfPrecisionPlugin` class as the parent function in lightning does not have it anymore.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -3603,6 +3603,7 @@ assert_frame_equal(training_curve, gt_curve, rtol=1e-3, atol=1e-3)"'''
         trainer.precision=16 \
         trainer.gradient_clip_val=1.0 \
         exp_manager.exp_dir=examples/nlp/language_modeling/gpt_pretrain_results \
+        exp_manager.resume_if_exits=False \
```
Comment (titu1994) [inline Jenkinsfile:3606]: resume_if_exists*

Comment (athitten) [reply Jenkinsfile:3606]: Thanks @titu1994, corrected the typo. Also the reason behind setting this to False is that this test aka `Megatron GPT Pretraining and Resume Training PP=2` is picking up the last checkpoint from the previous test aka `Megatron GPT with KERPLE Pretraining and Resume Training TP=2` and it errors out as it tries to find that checkpoint in the current test's `exp_dir`, but can't find it.

Comment (athitten) [reply Jenkinsfile]: This was happening bcoz I had commented these lines out by mistake in the lightning 2.0 upgrade PR: `https://github.com/NVIDIA/NeMo/blob/main/Jenkinsfile#L3583-L3584`. Fixing it ASAP in a separate PR.

**Review by ericharper — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**